### ISSUE TITLE
mid-rolls freezes in 3.2.15 if multiple providers are used

### ISSUE THREAD

**Issue by ViliusS** [flowplayer/flash#42]
Test case http://post.lnk.lt/flow/midroll3215_bad.html As you can see we have tried to use Flowplayer 3.2.15 with OVA plugin and after the mid-roll in 15th second the playout just freezes.

We have found that adding:

providers: {
 http: 'http',
 rtmp: 'rtmp'
}

to the OVA config solves the problem -> http://post.lnk.lt/flow/midroll3215_good.html . However this should not be needed as these providers are the default.

This is a regression from 3.2.14 which works just fine without any code additions. See http://post.lnk.lt/flow/midroll3214_good.html

**Comment by phloxic** [comment]
This seems to change with ova implementations. I definitely remember I had to do this with some previous version of ova.

**Comment by shishir12** [comment]
We have tried with the above solution, which still dint solve our problem

**Comment by ViliusS** [comment]
Can you check if 3.2.14 has the same problem?

**Comment by danrossi** [comment]
i dont get ads after 15 seconds. again have you reported these issues with the ova support ? ill email paul about these also.

**Comment by phloxic** [comment]
See #46 - ova + mid-roll + flowplayer-3.2.15 is broken (regardless of provider switch) - http://flash.flowplayer.org/forum/5/104191#post-104235

**Comment by ViliusS** [comment]
I have already contacted OVA support. Hopefully will get their response soon.

Strange that you don't get ad at 15 seconds. I just tried couple of times and it works.

**Comment by danrossi** [comment]
I think it's a region thing perhaps, im not in europe.

**Comment by danrossi** [comment]
@ shishir12 your problem is unrelated, I dont have the sources handy but it looks related to a problem with ova within one of the clip events something it needs is empty therefore an error.

**Comment by ViliusS** [comment]
@danrossi can you check if you can get http://lt.adocean.pl/files/oohefmenoi/ubqqhsgonq/R16796.mp4 to open?

**Comment by danrossi** [comment]
not in chrome / osx most likely a mimetypes issue

HTTP/1.1 200 OK
Accept-Ranges: none
Date: Fri, 26 Oct 2012 08:49:21 GMT
Cache-Control: public, must-revalidate, max-age=4320000
Last-Modified: Mon, 13 Aug 2012 12:11:17 GMT
ETag: "5028EEE5001655B8880160E8"
Expires: Sat, 15 Dec 2012 08:49:21 GMT
P3P: CP="NOI DSP COR NID PSAo OUR IND"
Connection: close
Content-Length: 1463736

**Comment by danrossi** [comment]
https://dl.dropbox.com/u/3394987/flowplayer-3.2.15.zip Please test if this works.

**Comment by ViliusS** [comment]
The mid-roll doesn't freeze anymore. That's great! However the control bar doesn't switch to the main video scale after mid-roll. For example main video is ~19 minutes long, the mid-roll is 10 seconds long. Control bar scale will remain at 10 seconds after main video is resumed. See http://post.lnk.lt/flow/dev/mid/controlbar.png what I mean. I have moved a mouse near the end of control bar and it shows 00:09 seconds. Also nothing happens if I try to seek forward.

I sure I've seen this bug before very rarely. However now it's constant.

**Comment by danrossi** [comment]
Just to confirm neither the original code or this problem appears with a standard instream configuration, so its an imcompatibikity with ova. How do I get a copy to test it ?

**Comment by danrossi** [comment]
clip: {

```
        // baseUrl for both parent and instream clips
        baseUrl: 'http://stream.flowplayer.org',

        // duration for the parent clip
        duration: 10,

        // instream playlist
        playlist: [

            // in the middle of the parent clip (ie. "mid-roll") starting at
            // 5 seconds
            {
                url: 'honda_accord.flv', duration: 5, position: 2,
                //controls: {all: false, volume: true, fullscreen: true}
            }
        ]
}
```

this config functions fine

**Comment by danrossi** [comment]
The other problem is an issue somewhere else.

**Comment by ViliusS** [comment]
For testing purposes you can load OVA directly from this link http://post.lnk.lt/flow/dev/mid/ova.swf

My test case: http://post.lnk.lt/flow/dev/mid/test.html

**Comment by danrossi** [comment]
I got your test working although having some latency issues playing it back. I can identify both errors and will look to see if I can fix them thanks.

**Comment by danrossi** [comment]
ok just a heads up, the problem is its looking for a nextclip and there is no nextclip because its an instream clip, so I think it's fairl to just close the stream to clean things up and return back to the parent clip.

**Comment by danrossi** [comment]
https://dl.dropbox.com/u/3394987/flowplayer.commercial-3.2.15.zip this should fix the first problem.

**Comment by ViliusS** [comment]
The mid-roll freezes in this version again :( I have updated my example http://post.lnk.lt/flow/dev/mid/test.html

**Comment by danrossi** [comment]
https://dl.dropbox.com/u/3394987/flowplayer.commercial-3.2.15.zip It probably didn't zip properly.

**Comment by ViliusS** [comment]
Yep, now it works.

**Comment by danrossi** [comment]
https://dl.dropbox.com/u/3394987/flowplayer.controls-3.2.14.zip
https://dl.dropbox.com/u/3394987/flowplayer.commercial-3.2.15.zip

The controlbar problem is when it returns back to the main clip it resumes but the time updater doesn't start so I fixed that and it will reposition the scrubber buffers and duration.

**Comment by ViliusS** [comment]
Thank you. Confirming as fixed.

**Comment by danrossi** [comment]
reopning for testing

**Comment by ViliusS** [comment]
GIven that version 3.2.15 contains some high profile bugs, can we expect bug fixes merged and 3.2.16 released soon? Seeing how Flowplayer5 gets new release every week, it takes really too long for simple bugfix Flash version to be appear.

**Comment by danrossi** [comment]
Hi you should have a working player right now right ? Im having to personally debug the ova plugin still because paul can't figure it out so I haven't got around to it.

**Comment by ViliusS** [comment]
Yes we have. However we cannot use it in production because of company policy. We can only use officialy released stable software. We are fine to wait for the OVA fix until the version _after_ that, but current fixes with freezing mid-rolls and double firing onBegin events are needed badly :(

**Comment by ViliusS** [comment]
Will it be faster if we would bought a support ticket? If yes, just tell us, it's not really a problem for us.

I have a deadline in front of me set by the management and if we won't have a production release ready before the end of this year, we will have no choice but to migrate to JWPlayer (even if I like Flowplayer a lot more personally).

**Comment by ViliusS** [comment]
Confirmed as fixed in 3.2.16.
